lib/semantics/check-stop.cc

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2019, Arm Ltd.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "check-stop.h"
+#include "semantics.h"
+#include "tools.h"
+#include "../common/Fortran.h"
+#include "../evaluate/expression.h"
+#include "../parser/parse-tree.h"
+#include <optional>
+
+void Fortran::semantics::StopChecker::Enter(
+    const Fortran::parser::StopStmt &stmt) {
+  const auto &sc{std::get<std::optional<Fortran::parser::StopCode>>(stmt.t)};
```

Comment:
We prefer to spell things out, like stopCode, but not like Stmt or Msg.

Reply:
you meant stopStmt instead of stmt?

Reply:
I think he means `stopCode` instead of `sc`.

Reply:
Done.